addons/account/migrations/9.0.1.1/post-migration.py

```diff
@@ -0,0 +1,362 @@
+# -*- coding: utf-8 -*-
+# © 2016 Serpent Consulting Services Pvt. Ltd.
+# © 2016 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+import logging
+from openupgradelib import openupgrade
+logger = logging.getLogger('OpenUpgrade')
+from openerp.modules.registry import RegistryManager
+
+def map_bank_state(cr):
+    openupgrade.map_values(
+        cr,
+        openupgrade.get_legacy_name('state'), 'state',
+        [('draft', 'open')],
+        table='account_bank_statement', write='sql')
+
+def map_type_tax_use(cr):
+    openupgrade.map_values(
+        cr,
+        openupgrade.get_legacy_name('type_tax_use'), 'type_tax_use',
+        [('all', 'none')],
+        table='account_tax', write='sql')
+
+def map_type_tax_use_template(cr):
+    openupgrade.map_values(
+        cr,
+        openupgrade.get_legacy_name('type_tax_use'), 'type_tax_use',
+        [('all', 'none')],
+        table='account_tax_template', write='sql')
+
+def map_journal_state(cr):
+    openupgrade.map_values(
+        cr,
+        openupgrade.get_legacy_name('type'), 'type',
+        [('purchase_refund', 'purchase'), ('sale_refund', 'sale'), ('situation', 'general')],
+        table='account_journal', write='sql')
+
+def account_templates(cr):
+
+    # Making Account Chart Template Company-Specific
+
+    cr.execute("""
+    select id from account_chart_template
+    """)
+    chart_ids = cr.dictfetchall()
+    
+    cr.execute("""
+    select id from res_company
+    """)
+    company_ids = cr.dictfetchall()
+    list_of_companies = company_ids[1:]
+    
+    cr.execute("""
+    select count(id) from res_company
+    """)
+    company = cr.fetchone()
+    no_of_companies = int(company[0])
+    company_count = no_of_companies - 1
+    
+    cr.execute("""
+    select count(id) from account_chart_template
+    """)
+    chart = cr.fetchone()
+    chart_template_count = int(chart[0])
+    
+    cr.execute("""
+    UPDATE account_chart_template SET company_id = r.id 
+    from res_company r
+    """)
+
+    for n in range(company_count):
+        company_id = list_of_companies[n]['id']
+        for m in range(chart_template_count):
+            chart_id = chart_ids[m]['id']
+            cr.execute("""
```

Comment:
I don't think you should duplicate all the templates here. The important part is that we *assign* some existing template to the company, and the tricky part is to assign the right one (based on the one initially used). I'll come up with a PR on yours soon